tests/unit/test_state.py

```diff
@@ -190,6 +223,25 @@ def test_config_backends_invalid_backends(invalid_backends: str, error_message:
     assert str(err.value) == error_message
 
 
+def test_http_protocol():
+    """
+    arrange: Mock charm with valid configurations.
+    act: Use a http as protocol, and create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROTOCOL_CONFIG_NAME] = "http"
+
+    config = Configuration.from_charm(charm)
+    assert config.hostname == "example.com"
```

Comment:
nit: have a newline to separate the act from the assert block

Reply:
Update the file to comply with this.